ui/tests/integration/components/pki/pki-key-parameters-test.js

```diff
@@ -0,0 +1,98 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, click, fillIn } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { setupEngine } from 'ember-engines/test-support';
+import { SELECTORS } from 'vault/tests/helpers/pki-engine';
+
+module('Integration | Component | pki-key-parameters', function (hooks) {
+  setupRenderingTest(hooks);
+  setupEngine(hooks, 'pki');
+
+  hooks.beforeEach(function () {
+    this.store = this.owner.lookup('service:store');
+    this.model = this.store.createRecord('pki/pki-role-engine');
+    this.model.backend = 'pki';
+  });
+
+  test('it should render the component and display the correct defaults', async function (assert) {
+    assert.expect(3);
+    await render(
+      hbs`
+      <div class="has-top-margin-xxl">
+        <PkiKeyParameters
+          @model={{this.model}}
+          @group="Key parameters"
+        />
+       </div>
+  `,
+      { owner: this.engine }
+    );
+    await click(SELECTORS.keyParams);
+    assert.dom(SELECTORS.keyType).hasValue('rsa');
+    assert.dom(SELECTORS.keyBits).hasValue('2048');
+    assert.dom(SELECTORS.signatureBits).hasValue('0');
+  });
+
+  test('it should set the model properties of key_type and key_bits when key_type changes', async function (assert) {
+    assert.expect(11);
+    await render(
+      hbs`
+      <div class="has-top-margin-xxl">
+        <PkiKeyParameters
+          @model={{this.model}}
+          @group="Key parameters"
+        />
+       </div>
+  `,
+      { owner: this.engine }
+    );
+    assert.deepEqual(this.model.keyType, 'rsa', 'sets the default value for key_type on the model.');
+    assert.deepEqual(this.model.keyBits, 2048, 'sets the default value for key_bits on the model.');
+    assert.deepEqual(this.model.signatureBits, 0, 'sets the default value for signature_bits on the model.');
```

Comment:
I think you want `strictEqual` here. `deepEqual` is to compare objects. You could also use `propEqual` to bundle into one assertion:
```js
assert.propEqual(this.model,
  {
    keyType: 'rsa',
    keyBits: 2048,
    signatureBits: 0,
  },
  'sets default model values'
);
```

Reply:
Good catch! Thank you.

Reply:
And very cool about propEqual!